packages/api/config.mts

```diff
@@ -44,12 +44,35 @@ export async function updateConfig(attrs: Partial<Config>) {
   return db.update(configs).set(attrs).returning();
 }
 
-export async function getSecrets(): Promise<Record<string, string>> {
-  const results = await db.select().from(secrets);
-  return results.reduce((acc: Record<string, string>, { name, value }) => {
-    acc[name] = value;
-    return acc;
-  }, {});
+export async function getSecrets(): Promise<Array<SecretWithAssociatedSessions>> {
+  const secretsResult = await db.select().from(secrets);
+  const secretsToSessionResult = await db.select()
+    .from(secretsToSession)
+    .where(
+      inArray(
+        secretsToSession.secret_id,
+        secretsResult.map(secret => secret.id),
+      ),
+    );
+
+  return secretsResult.map(secret => ({
+    name: secret.name,
+    value: secret.value,
+    associatedWithSessionIds: (
+      secretsToSessionResult
+        .filter(secretToSession => secretToSession.secret_id === secret.id)
+        .map(secretToSession => secretToSession.session_id)
+    ),
+  }));
+}
```

Comment:
I opted to return a list of associated sessions in the root `/secrets` api response and use this same request to power the new secrets panel, which required some reshuffling of the response format.